hnn_core/tests/test_network.py

```diff
@@ -85,32 +85,57 @@ def test_spikes(tmpdir):
+    spike_times = [[2.3456, 7.89], [4.2812, 93.2]]
+    spike_gids = [[1, 3], [5, 7]]
+    spike_types = [['L2_pyramidal', 'L2_basket'],
+                   ['L5_pyramidal', 'L5_basket']]
     tstart, tstop = 0.1, 98.4
     gid_dict = {'L2_pyramidal': range(1, 2), 'L2_basket': range(3, 4),
                 'L5_pyramidal': range(5, 6), 'L5_basket': range(7, 8)}
-    spikes = Spikes(times=spiketimes, gids=spikegids, types=spiketypes)
+    spikes = Spikes(spike_times=spike_times, spike_gids=spike_gids,
+                    spike_types=spike_types)
     spikes.plot_hist(show=False)
     spikes.write(tmpdir.join('spk_%d.txt'))
     assert spikes == read_spikes(tmpdir.join('spk_*.txt'))
 
     assert ("Spikes | 2 simulation trials" in repr(spikes))
 
-    with pytest.raises(TypeError, match="times should be a list of lists"):
-        spikes = Spikes(times=([2.3456, 7.89], [4.2812, 93.2]), gids=spikegids,
-                        types=spiketypes)
+    with pytest.raises(TypeError,
+                       match="spike_times should be a list of lists"):
+        spikes = Spikes(spike_times=([2.3456, 7.89], [4.2812, 93.2]),
+                        spike_gids=spike_gids, spike_types=spike_types)
 
-    with pytest.raises(TypeError, match="times should be a list of lists"):
-        spikes = Spikes(times=[1, 2], gids=spikegids, types=spiketypes)
+    with pytest.raises(TypeError,
+                       match="spike_times should be a list of lists"):
+        spikes = Spikes(spike_times=[1, 2], spike_gids=spike_gids,
+                        spike_types=spike_types)
 
-    with pytest.raises(ValueError, match="times, gids, and types should be "
-                       "lists of the same length"):
-        spikes = Spikes(times=[[2.3456, 7.89]], gids=spikegids,
-                        types=spiketypes)
+    with pytest.raises(ValueError, match="spike times, gids, and types should "
+                       "be lists of the same length"):
+        spikes = Spikes(spike_times=[[2.3456, 7.89]], spike_gids=spike_gids,
+                        spike_types=spike_types)
 
-    spikes = Spikes(times=spiketimes, gids=spikegids, types=spiketypes)
+    spikes = Spikes(spike_times=spike_times, spike_gids=spike_gids,
+                    spike_types=spike_types)
+
+    with pytest.raises(TypeError, match="indices must be int, slice, or "
+                       "array-like, not str"):
+        spikes['1']
+
+    with pytest.raises(TypeError, match="indices must be int, slice, or "
+                       "array-like, not float"):
+        spikes[1.0]
+
+    with pytest.raises(ValueError, match="ndarray cannot exceed 1 dimension"):
+        spikes[np.array([[1, 2], [3, 4]])]
+
+    with pytest.raises(TypeError, match="gids must be of dtype int, "
+                       "not float64"):
+        spikes[np.array([1, 2, 3.0])]
+
+    with pytest.raises(TypeError, match="gids must be of dtype int, "
+                       "not float64"):
+        spikes[[0, 1, 2, 2.0]]
```

Comment:
are there any tests that check the dimensions of `vsoma` and if they match the number of trials and simulation length etc?